### PR TITLE
EFA Integration Test

### PR DESCRIPTION
**Pull request by sean-smith**
Tests that EFA is correctly enabled:

**Only review**
tests/integration-tests/tests/test_efa/test_efa.py
tests/integration-tests/tests/test_efa/test_efa/test_efa/pcluster.config.ini

### EFA Enabled
```bash
00:00.0 Host bridge: Intel Corporation 440FX - 82441FX PMC [Natoma]
00:01.0 ISA bridge: Intel Corporation 82371SB PIIX3 ISA [Natoma/Triton II]
00:01.3 Non-VGA unclassified device: Intel Corporation 82371AB/EB/MB PIIX4 ACPI (rev 08)
00:03.0 VGA compatible controller: Amazon.com, Inc. Device 1111
00:04.0 Non-Volatile memory controller: Amazon.com, Inc. Device 8061
00:05.0 Ethernet controller: Amazon.com, Inc. Elastic Network Adapter (ENA)
00:06.0 Ethernet controller: Amazon.com, Inc. Device efa0
```

### Non-EFA

```bash
$ lspci
00:00.0 Host bridge: Intel Corporation 440FX - 82441FX PMC [Natoma]
00:01.0 ISA bridge: Intel Corporation 82371SB PIIX3 ISA [Natoma/Triton II]
00:01.3 Non-VGA unclassified device: Intel Corporation 82371AB/EB/MB PIIX4 ACPI (rev 08)
00:03.0 VGA compatible controller: Amazon.com, Inc. Device 1111
00:04.0 Non-Volatile memory controller: Amazon.com, Inc. Device 8061
00:05.0 Ethernet controller: Amazon.com, Inc. Elastic Network Adapter (ENA)
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/test_efa/test_efa.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import logging
+
+import pytest
+
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+
+
+@pytest.mark.regions(["us-east-1"])
+@pytest.mark.instances(["c5n.18xlarge", "p3dn.24xlarge", "i3en.24xlarge"])
```
Comment (demartinofra) [inline tests/integration-tests/tests/test_efa/test_efa.py]: just as a reminder: we should enable these instances in jenkins

File: tests/integration-tests/tests/test_efa/test_efa/test_efa/pcluster.config.ini [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+aws_region_name = {{ region }}
+
+[cluster default]
+base_os = {{ os }}
+key_name = {{ key_name }}
+vpc_settings = parallelcluster-vpc
+scheduler = {{ scheduler }}
+master_instance_type = t2.micro
+compute_instance_type = {{ instance }}
+initial_queue_size = 2
```
Comment (demartinofra) [inline tests/integration-tests/tests/test_efa/test_efa/test_efa/pcluster.config.ini]: why do we need 2 compute nodes?

File: tests/integration-tests/tests/test_efa/test_efa.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    _test_efa_installed(remote_command_executor)
+
+
+def _test_efa_installed(remote_command_executor):
+    # Output contains:
+    # 00:06.0 Ethernet controller: Amazon.com, Inc. Device efa0
+    logging.info("Testing EFA Installed")
+    version = remote_command_executor.run_remote_command("qrsh /sbin/lspci").stdout
```
Comment (demartinofra) [inline tests/integration-tests/tests/test_efa/test_efa.py]: this is testing efa on the master node. Shall we verify this on the compute node as well by running a job that checks lspci output?

Comment (sean-smith) [reply tests/integration-tests/tests/test_efa/test_efa.py]: this submits a job to one of the compute nodes

File: tests/integration-tests/tests/test_efa/test_efa/test_efa/pcluster.config.ini (hunk 2 above)
Comment (sean-smith) [reply tests/integration-tests/tests/test_efa/test_efa/test_efa/pcluster.config.ini]: To run an MPI job that communicates between nodes

**Pull request closed without merging**